### PR TITLE
add gdptools

### PR DESCRIPTION
**Pull request by ocefpaf**
This one falls under the exception in CFEP-05 b/c it won't see a "stable" release anytime soon, even though it is a stable release.

See https://github.com/conda-forge/cfep/blob/main/cfep-05.md#exceptions

### CODE REVIEW

**Comment by ocefpaf** [comment]
@conda-forge/staged-recipes this one is ready for review!

**Comment by ocefpaf** [comment]
@conda-forge/staged-recipes this one is ready for review!

File: recipes/gdptools/tests/test_RWE_Class.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+from gdptools.ancillary import _generate_weights_pershp
+from gdptools.ancillary import _get_cells_poly
+from gdptools.ancillary import _get_crs
+from gdptools.ancillary import _get_data_via_catalog
+from gdptools.ancillary import _get_shp_file
+from gdptools.gdp_data_class import CatGrids
+from gdptools.gdp_data_class import CatParams
+
+
+@pytest.mark.parametrize(
```
Comment (msarahan) [inline recipes/gdptools/tests/test_RWE_Class.py:23]: in the longer-term, these should probably move to the project instead of this recipe, right?

Comment (ocefpaf) [reply recipes/gdptools/tests/test_RWE_Class.py:23]: Yep. I'm planning on sending a PR upstream so we can remove this from here.

File: recipes/gdptools/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+test:
+  files:
+   - tests/
+  requires:
+    - pip
+    - pytest
+  imports:
+    - gdptools
+  commands:
+    # Some of the pins are too aggressive making the package impossible to install.
```
Comment (msarahan) [inline recipes/gdptools/meta.yaml:63]: The pins are terribly tight. Would it be ok to remove some of the upper bounds? I guess you really already have on several of them.

I've been advising people to not have upper bounds unless they really know they need them. Maybe I'm wrong on that. I feel like here at conda-forge, we have repodata patching to fix/impose upper bounds when needed, and we should not prematurely introduce conflicts that don't really exist.

File: recipes/gdptools/tests/test_RWE_Class.py (hunk 1 above)
Comment (ocefpaf) [reply recipes/gdptools/tests/test_RWE_Class.py:23]: That is why I added the tests. I relaxed all those that I could and ensure a passing build. The others will need some upstream modifications.

Yeah, folks usually confuse apps with libraries.

File: recipes/gdptools/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/gdptools/meta.yaml:63]: That is why I added the tests. I relaxed all those that I could and ensure a passing build. The others will need some upstream modifications.

Yeah, folks usually confuse apps with libraries.

**Pull request merged**